core/src/main/java/org/apache/calcite/plan/volcano/RelSubset.java

```diff
@@ -128,6 +129,16 @@
    */
   private boolean enforceDisabled = false;
 
+  /**
+   * the upper bound of the last OptimizeGroup call
+   */
+  RelOptCost upperBound;
+
+  /**
+   * RelNode ids that is invoked passThrough method before
+   */
+  Set<RelNode> passThroughCache;
```

Comment:
Correct the comment too~

Reply:
Will do it.

Reply:
Hmm. I think it is better to store the passThroughCache to RelSet, where the key is the traitset, the value are relnodes.

Reply:
I see no differences. Currently, RelSet + Traits = RelSubset